src/containerapp/azext_containerapp/custom.py

```diff
@@ -448,11 +449,22 @@ def create_containerapp(cmd,
                         registry_identity=None,
                         workload_profile_name=None,
                         termination_grace_period=None,
-                        secret_volume_mount=None):
+                        secret_volume_mount=None,
+                        repo=None,
+                        token=None,
+                        branch=None,
+                        context_path=None,
+                        service_principal_client_id=None,
+                        service_principal_client_secret=None,
+                        service_principal_tenant_id=None):
+    from ._up_utils import (_validate_create_args,_reformat_image,_get_dockerfile_content, _get_ingress_and_target_port,ContainerApp,ResourceGroup,ContainerAppEnvironment, _get_registry_details, _create_github_action,
+                            get_token, _has_dockerfile)
+    from ._github_oauth import cache_github_token
     register_provider_if_needed(cmd, CONTAINER_APPS_RP)
     validate_container_app_name(name, AppType.ContainerApp.name)
     validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait)
     validate_revision_suffix(revision_suffix)
+    _validate_create_args(cmd, source, repo, registry_server, registry_user, registry_pass)
```

Comment:
Please rebase newest code and move following validate logic to ContainerAppCreateDecorator.validate_arguments

`_validate_create_args(cmd, source, repo, registry_server, registry_user, registry_pass)`

Reply:
Merged latest changes from main